Automattic/wp-calypso
# Update/legacy to recommended map 2

#### Proposed Changes

* this change adds on to the PR to display plan recommendations to users who try to add a legacy plan to the cart.  Ref https://github.com/Automattic/wp-calypso/tree/fix/missing-plan-recommendations-on-pricing-page
* The plan recommendation page would display no-longer-available products such as backup daily
* This updates what is suggested based on https://jetpack.com/support/jetpack-plan-equivalency-chart/
* This will also render the button for the deprecated plan's card as disabled

![Screen Shot 2022-11-09 at 10 03 18 AM](https://user-images.githubusercontent.com/12895386/200865477-5697ede9-1daa-4c08-9980-fa31d797ed0b.png)


#### Testing Instructions

* Use the calypso.live link below
* Try to visit the checkout page with an old product by appending one of these to the live link:
* /checkout/:site/jetpack_personal
* /checkout/:site/jetpack_premium
* /checkout/:site/jetpack_business
(substitute :site with a valid Jetpack site you own)
* You should see a section above the current Pricing page that informs you the selected plan is not available and offers alternatives:
* Personal should show Backup + Anti Spam
* Premium should show Security + Videopress
* Business/Professional should show Complete
* Verify that you can add the products to the cart when you click on the CTA button
* Verify the button for the current (deprecated) product is disabled

#### Pre-merge Checklist






Related to #
